modules/virt-measuring-latency-vm-secondary-network.adoc

```diff
@@ -0,0 +1,251 @@
+kind: ClusterRole
+metadata:
+  name: kiagnose
+rules:
+  - apiGroups: [ "" ]
+    resources: [ "configmaps" ]
+    verbs:
+      - get
+      - list
+      - create
+      - update
+      - patch
+  - apiGroups: [ "" ]
+    resources: [ "namespaces" ]
+    verbs:
+      - get
+      - list
+      - create
+      - delete
+      - watch
+  - apiGroups: [ "" ]
+    resources: [ "serviceaccounts" ]
+    verbs:
+      - get
+      - list
+      - create
+  - apiGroups: [ "rbac.authorization.k8s.io" ]
+    resources:
+      - roles
+      - rolebindings
+      - clusterrolebindings
+    verbs:
+      - get
+      - list
+      - create
+      - delete
+  - apiGroups: [ "rbac.authorization.k8s.io" ]
+    resources:
+      - clusterroles
+    verbs:
+      - get
+      - list
+      - create
+      - bind
+  - apiGroups: [ "batch" ]
+    resources: [ "jobs" ]
+    verbs:
+      - get
+      - list
+      - create
+      - delete
+      - watch
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: kiagnose
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: kiagnose
+subjects:
+  - kind: ServiceAccount
+    name: kiagnose
+    namespace: kiagnose
+...
+----
+====
+
+. Apply the framework manifest:
++
+[source,terminal]
+----
+$ oc apply -f <framework-manifest>.yaml
+----
+
+. Create a configuration file that contains the `ClusterRole` and `Role` objects with permissions that are required by the checkup to access the cluster:
++
+.Example checkup role manifest file
+[source,yaml]
+----
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: kubevirt-vm-latency-checker
+rules:
+- apiGroups: ["kubevirt.io"]
+  resources: ["virtualmachineinstances"]
+  verbs: ["get", "create", "delete"]
+- apiGroups: ["subresources.kubevirt.io"]
+  resources: ["virtualmachineinstances/console"]
+  verbs: ["get"]
+- apiGroups: ["k8s.cni.cncf.io"]
+  resources: ["network-attachment-definitions"]
+  verbs: ["get"]
+----
+
+. Apply the checkup roles manifest:
++
+[source,terminal]
+----
+$ oc apply -f <latency-roles>.yaml
+----
+
+. Create a `ConfigMap` manifest that contains the input parameters for the checkup. The config map provides the input for the framework to run the checkup and also stores the results of the checkup.
++
+.Example input config map
+[source,yaml]
+----
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: kubevirt-vm-latency-checkup
+  namespace: kiagnose
+data:
+  spec.image: registry.redhat.io/container-native-virtualization/vm-network-latency-checkup:v4.11.0
+  spec.timeout: 10m
+  spec.clusterRoles: |
+    kubevirt-vmis-manager
+  spec.param.network_attachment_definition_namespace: "default" <1>
+  spec.param.network_attachment_definition_name: "bridge-network" <2>
+  spec.param.max_desired_latency_milliseconds: "10" <3>
+  spec.param.sample_duration_seconds: "5" <4>
+----
+<1> The namespace where the `NetworkAttachmentDefinition` object resides.
```

Comment:
This is the first mention of a required NetworkAttachmentDefinition for this tests.
I think this page's Prerequisites section should mention that this checkup requires having a NetworkAttachmentDefinition attached to a bridge device, which will be used by the checkup VMIs.
@EdDev wdyt?

Reply:
The NAD can be specified as a per-requisite for running the checkup (this latency specific one).
But there is no requirement for it to be using `bridge`. We support for example `SR-IOV` which is something else, and that is our main case in practice.

Reply:
I've just seen the old commit mentioning a bridge nad. I agree, no need for a bridge NAD, but just saying a NAD is needed should be enough